test/integration/resourcemanager/managedresource/resource_test.go

```diff
@@ -245,22 +248,56 @@ var _ = Describe("ManagedResource controller tests", func() {
 			})
 		})
 
+		Describe("resource data changes", func() {
+			BeforeEach(func() {
+				configMap.Data = map[string]string{"foo": "bar"}
+				secretForManagedResource.Data = secretDataForObject(configMap, dataKey)
+			})
+
+			It("should set ManagedResource to unhealthy", func() {
+
+				patch := client.MergeFrom(secretForManagedResource.DeepCopy())
+				secretForManagedResource.Data = secretDataForObject(configMap, dataKey)
+				Expect(testClient.Patch(ctx, secretForManagedResource, patch)).To(Succeed())
+
+				Eventually(func(g Gomega) []gardencorev1beta1.Condition {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(Succeed())
+					return managedResource.Status.Conditions
+				}).Should(
+					containCondition(ofType(resourcesv1alpha1.ResourcesApplied), withStatus(gardencorev1beta1.ConditionTrue), withReason(resourcesv1alpha1.ConditionApplySucceeded)),
+					containCondition(ofType(resourcesv1alpha1.ResourcesHealthy), withStatus(gardencorev1beta1.ConditionUnknown), withReason(resourcesv1alpha1.ConditionChecksPending)),
+					containCondition(ofType(resourcesv1alpha1.ResourcesProgressing), withStatus(gardencorev1beta1.ConditionUnknown), withReason(resourcesv1alpha1.ConditionChecksPending)),
+				)
+
+				Consistently(func(g Gomega) map[string]string {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(configMap), configMap)).To(Succeed())
+					return configMap.Data
+				}).Should(HaveKeyWithValue("foo", "bar"))
+			})
+		})
+
 		Describe("new resource added", func() {
 			It("should successfully create a new resource", func() {
 				patch := client.MergeFrom(secretForManagedResource.DeepCopy())
 				secretForManagedResource.Data[newDataKey] = secretDataForObject(newResource, newDataKey)[newDataKey]
 				Expect(testClient.Patch(ctx, secretForManagedResource, patch)).To(Succeed())
 
-				Eventually(func(g Gomega) {
-					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(configMap), configMap)).To(Succeed())
-					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(newResource), newResource)).To(Succeed())
-				}).Should(Succeed())
+				Eventually(func(g Gomega) map[string][]byte {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(secretForManagedResource), secretForManagedResource)).To(Succeed())
+					return secretForManagedResource.Data
+				}).Should(HaveKey(newDataKey))
+				Eventually(func(g Gomega) map[string][]byte {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(secretForManagedResource), secretForManagedResource)).To(Succeed())
+					return secretForManagedResource.Data
+				}).Should(HaveKey(dataKey))
```

Comment:
What are you trying to achieve with these assertions?
I don't think it makes much sense to check that the secret is present that the test code has just created before. The secret itself is not even touched by the tested code.

Reply:
My bad, I have reverted back the changes.